go/vt/vtgate/endtoend/misc_test.go

```diff
@@ -0,0 +1,184 @@
+	}
+
+	//Truncate all the data and verify table is empty
+	exec(t, conn, "truncate user")
+	qr = exec(t, conn, "select * from user")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+}
+
+func TestScatterLimit(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	//Insert few records in the table
+	exec(t, conn, "begin")
+	exec(t, conn, "insert into user(id, name) values(1,'john'), (2, 'mark'), (3, 'paul'), (4, 'doug')")
+	exec(t, conn, "commit")
+
+	// Verify LIMIT works
+	qr := exec(t, conn, "select id from user order by id limit 2")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(1)] [INT64(2)]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	// Verify LIMIT with OFFSET works
+	qr = exec(t, conn, "select id from user order by id limit 1 offset 2")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(3)]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	// Verify when limit is greater than values in the table
+	qr = exec(t, conn, "select id from user order by id limit 100 offset 1")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(2)] [INT64(3)] [INT64(4)]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+}
+
+func TestVindexFunc(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// Test Vindex Function
+	qr := exec(t, conn, "select id, keyspace_id from hash where id = 1")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[VARBINARY("1") VARBINARY("\x16k@\xb4J\xbaK\xd6")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+}
+
+func TestAnalyzeTable(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// Test Analyze Table
+	qr := exec(t, conn, "analyze table t1")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[VARCHAR("vt_ks_-80.t1") VARCHAR("analyze") VARCHAR("status") TEXT("OK")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	// Test Analyze Table 2
+	qr = exec(t, conn, "analyze table t4_music_lookup")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[VARCHAR("vt_ks_-80.t4_music_lookup") VARCHAR("analyze") VARCHAR("status") TEXT("OK")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+}
+
+func TestTransactionModes(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// cfg.TransactionMode = vtgatepb.TransactionMode_SINGLE
+	//extra_args=['-transaction_mode', 'TWOPC'])
+	// fmt.Println(cluster.Config.TransactionMode)
+
+	//Insert trageted to multiple tables should fail with SINGLE trx mode
+	cluster.Config.TransactionMode = "SINGLE"
```

Comment:
@deepthi I was not able to find a good way to update the TransactionMode for a VtGate connection. Can you help me with that?

By looking at the code it should have happened if we update the cluster.Config.TransactionMode value.

Reply:
It seems like cluster.Config is only read while starting the VtComboProcess:
https://github.com/vitessio/vitess/blob/master/go/vt/vttest/local_cluster.go#L252

So I don't think it can be changed on the fly like this.

I took a quick look at the python code, but it isn't obvious how this is being done there. You will need to dig into the cursor code in https://github.com/vitessio/vitess/blob/master/py/vtdb/vtgate_cursor.py